### PR TITLE
Convert ews-build.webkit.org to multi-master (to improve responsiveness)

### PR DESCRIPTION
**Pull request by aj062**
#### 8df1e60d240a30bf4dfc17a6b32663ba946752ab
<pre>
Convert ews-build.webkit.org to multi-master (to improve responsiveness)
<a href="https://bugs.webkit.org/show_bug.cgi?id=268982">https://bugs.webkit.org/show_bug.cgi?id=268982</a>

Reviewed by Ryan Haddad.

* Tools/CISupport/ews-build/master.cfg:
* Tools/CISupport/ews-build/loadConfig.py:
* Tools/CISupport/ews-build-webserver/buildbot.tac: Copied from Tools/CISupport/ews-build/buildbot.tac.
* Tools/CISupport/ews-build-webserver/master.cfg: Copied from Tools/CISupport/ews-build/master.cfg.

Canonical link: <a href="https://commits.webkit.org/274295@main">https://commits.webkit.org/274295@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/b9d05d86077d0f805ecf195cab714d170960b083

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/38636 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/17566 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/40962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/41167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/34295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/40941 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/20370 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/14913 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/41167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/39209 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/20370 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/40962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/41167 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/20370 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/40962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/42442 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/20370 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/40962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/42442 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13452 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/14913 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/42442 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🧪 services~~](https://ews-build.webkit.org/#/builders/28/builds/38685 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/15064 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/40962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/13910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5030 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/14540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |

### CODE REVIEW

File: Tools/CISupport/ews-build-webserver/master.cfg [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    # See https://docs.buildbot.net/current/manual/configuration/www.html#example-configs
+    authz = util.Authz(
+        allowRules=[util.AnyEndpointMatcher(role='Buildbot-Administrators', defaultDeny=False),
+                    util.RebuildBuildEndpointMatcher(role='Contributors'),
+                    util.StopBuildEndpointMatcher(role='Committers'),
+                    util.AnyControlEndpointMatcher(role='Buildbot-Administrators')],
+        roleMatchers=[util.RolesFromGroups(groupPrefix='WebKit/')]
+    )
+    auth = util.GitHubAuth(GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET, apiVersion=4, getTeamsMembership=True, debug=True)
+    #c['www']['auth'] = auth
```
Comment (JonWBedard) [inline Tools/CISupport/ews-build-webserver/master.cfg]: Why are we commenting this out?

**Review by ryanhaddad — APPROVED** [review]

File: Tools/CISupport/ews-build-webserver/master.cfg [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import json
+import os
+import socket
+import sys
+
+from buildbot.plugins import reporters, util
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+loadConfig = importlib.import_module("ews-build.loadConfig")
```
Comment (JonWBedard) [inline Tools/CISupport/ews-build-webserver/master.cfg:11]: Much of this is copied from `ews-build`, right? Can we consider trying to share more of this code? I know in our other multi-master setups, duplication of code in `master.cfg` has been a source of problems.

File: Tools/CISupport/ews-build-webserver/master.cfg (hunk 1 above)
Comment (aj062) [reply Tools/CISupport/ews-build-webserver/master.cfg]: It was from my debugging, will remove. Thanks for noticing.

File: Tools/CISupport/ews-build-webserver/master.cfg (hunk 2 above)
Comment (aj062) [reply Tools/CISupport/ews-build-webserver/master.cfg:11]: Can probably look into de-duplication later-on. In our other multi-master setup, we never tried de-duplication in this file earlier. Also, I don't think duplication was a source of problem, the problem was specific things (like missing out required schedulers on one master).

**Review by JonWBedard — APPROVED** [review]

**Pull request merged**